parser/parser/cc/grammars/typedruby.ypp

```diff
@@ -1930,6 +1947,8 @@ array_premature_end: eof
                     {
                       driver.replace_stack($<stack>2);
                       $$ = driver.build.beginKeyword(self, $1, $3->body, $4);
+                      // TODO(jez) Make rewind_if_dedented aware of tBEFORE_EOF
+                      // TODO(jez) Use kend_or_eof in all rules that use rewind_if_dedented
```

Comment:
I'm going to defer this to a future PR to keep this PR small and make it easier to bisect/roll back in case that becomes necessary.